### DISCUSSION TITLE
Add record of consents

### DISCUSSION THREAD

**Discussion by nicola1989** [orestbida/cookieconsent#208] [General]
Hello, great cookie solution!
Is it possible to make sure that the consents are saved in a txt file, associated with an anonymized ip (e.g. 00.000.00.xx)?

In my state it is critical for compliance when there are marketing cookies.

Thanks in advance!

**Comment by tillsanders** [comment]
Hi @nicola1989,

since cookieconsent is a purely client side solution, there is no way to persist such information server side. You are, however, not the only one with this problem. See #171 for the current discussion on this matter.

Although there is no solution built-in, cookieconsent does provide methods that would allow integration with a server side component. You can find those in the readme.

Have a nice day ✌️

**Reply by regfullstack** [reply]
We definitely want to persist consent in our sql server database. Using cookieconesnt how would we do this? Anyway to call an api endpoint and send it relevant data after consent is given? What data is available once they consent? Ip address, etc.?

**Comment by helmar81** [comment]
Unfortunately, it is not working for me on my Astro.js project.
![Screenshot (355)](https://user-images.githubusercontent.com/42181669/218627162-9a20496d-fc3c-4b16-95ca-a2dd4644832b.png)

**Reply by orestbida** [reply]
Please open a new issue with a minimal reproducible example
